### ISSUE TITLE
No API method to programmatically clear column sort

### ISSUE THREAD

**Issue by nkoterba** [angular-ui/ui-grid#5207]
So I would like to clear the sorting on a column when a column is hidden by the user in the UI.

I have hooked into the `columnVisibilityChanged` event and now just need to "unsort" a specific column (the column that was just hidden) programmatically.

I couldn't find anything in the API that would allow me to do such a thing.

I found `vm.gridApi.grid.sortColumn(column, directionOrAdd, add)` but this doesn't support "clearing" a column's sort.

I thought something like `vm.gridApi.grid.sortColumn(column, false)` might work but all that does is clear any other sorted columns (in a multi-sort situation) and then sort my column in the next "sortDirectionCycle" (so if it was ASC, it goes to DESC; if it was DESC, it goes to null, etc.).

I can achieve what I need by doing the following:

``` javascript
    columnChanged.sort = {};
    sortChanged(vm.gridApi.grid, vm.gridApi.grid.getColumnSorting());
```

In short, I explicitly set the column's sort object to `{}` and then run my other code, but this exposes ui-grid internals...how ui-grid **resets** or **clears** sort is an implementation detail (setting it equal to {}) that I really shouldn't know about.

My suggestion would be to add a `removeSort` API method, or support passing a specific set of parameters into `sortColumn` that enables "clearing the sort on a column".

**Comment by PaulL1** [comment]
I think you can change the sort on the column defs, and that will then apply.

**Comment by nkoterba** [comment]
@PaulL1 Thanks for the quick reply.

A few comments on your suggestion:
1) It seems to me if there's an API call to sort a column it would be natural to have an API call to clear a column sort.
2) If I have a specific column that I need to unsort, it seems inefficient relying on angular's `$watch` of my `columnDefs` to trigger the UI update.

Setting `column.sort = {}` and then re-fetching server-side data does indeed have the effect I want and the UI state of the grid seems to correctly update.  I will try your solution above but anticipate it too will have the effect I am seeking.

**Comment by PaulL1** [comment]
Agree, it would be nice to have a clearer API in this area.  My suggestion was more to avoid the need to hit the grid internals quite so much - the properly abstracted way to do it is via columnDefs, as you note hitting the internals will also work.

**Comment by jainrahulkumar** [comment]
found this working resetColumnSorting([excludedColumn])
